libexec/python/base.py

```diff
@@ -337,6 +337,10 @@ def prepare_request(self, url, data=None, headers=None):
         else:
             request = Request(url=url,
                               headers=headers)
+
+            # Force User-Agent always
+            request.add_header('User-Agent', 'Singularity')
```

Comment:
It's customary to put the software version in a user agent. Is it possible/simple to do that?

Reply:
Not so easy I don't think... I looked how to get it in the Python.... and couldn't work it out in the time I was looking.